app/services/fetch_coronavirus_statistics_service.rb

```diff
@@ -0,0 +1,84 @@
+class FetchCoronavirusStatisticsService
+  CACHE_KEY = "coronavirus_statistics".freeze
+
+  Statistics = Struct.new(:cumulative_vaccinations,
+                          :cumulative_vaccinations_date,
+                          :hospital_admissions,
+                          :hospital_admissions_date,
+                          :new_positive_tests,
+                          :new_positive_tests_date,
+                          keyword_init: true)
+
+  def self.call
+    new.call
+  end
+
+  def call
+    # This API times out on occassion, so we'll keep a copy of our potentially
+    # stale data to use as a backup
+    backup_statistics = Rails.cache.fetch(CACHE_KEY)
+    statistics = fetch_current_statistics || backup_statistics
+
+    Statistics.new(statistics) if statistics && statistics.any?
+  end
+
+  private_class_method :new
+
+private
+
+  def fetch_current_statistics
+    Rails.cache.fetch(CACHE_KEY, expires_in: 30.minutes, race_condition_ttl: 5.minutes) do
+      request_statistics
+    end
+  rescue StandardError => e
+    Rails.logger.warn("Failed to load coronavirus statistics: #{e}")
+    intermittent_error = e.is_a?(Faraday::ConnectionFailed) ||
+      e.is_a?(Faraday::ServerError) && [502, 504].include?(e.response_status)
+
+    GovukError.notify(e) if !e.is_a?(Faraday::TimeoutError) && !intermittent_error
```

Comment:
The logic for `intermittent_error` is a bit of brain twister. Until I saw the tests, I read it to mean we notify Sentry if the error **is** a 502 or 504.

Reply:
Yup, you're right. I've refactored it to be an easier read.